compiler/src/dotty/tools/dotc/core/tasty/TastyPickler.scala

```diff
@@ -48,13 +42,12 @@ class TastyPickler(val rootCls: ClassSymbol) {
     val uuidHi: Long = otherSectionHashes.fold(0L)(_ ^ _)
 
     val headerBuffer = {
-      val buf = new TastyBuffer(header.length + TastyPickler.versionStringBytes.length + 32)
+      val buf = new TastyBuffer(header.length + TastyPickler.versionString.length + 32)
```

Comment:
this seems wrong - string length != utf-8 bytes length, e.g.
```scala
scala> val sc = "Scala 3.3.1➽"
val sc: String = Scala 3.3.1➽
                                                                                                                                                                                  
scala> sc.length
val res0: Int = 12
                                                                                                                                                                                  
scala> val scBytes = sc.getBytes(java.nio.charset.StandardCharsets.UTF_8)
val scBytes: Array[Byte] = Array(83, 99, 97, 108, 97, 32, 51, 46, 51, 46, 49, -30, -98, -67)
                                                                                                                                                                                  
scala> scBytes.length
val res1: Int = 14
```

Reply:
The `+ 32` covers a bit more than it needs. At least for the current version we do not have to relocate the buffer.

We also do not have an exact formula to know how much space the `Nat`s will take.

Reply:
I guess most of the time we shouldn't have these non ascii strings but :/

Reply:
That was my assumption.